python/python_direct/runtime.cpp

```diff
@@ -0,0 +1,144 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <bindings.h>
+#include <python_utils.h>
+
+#include <fusion.h>
+#include <runtime/executor_kernel_arg.h>
+#include <runtime/fusion_executor_cache.h>
+#include <runtime/fusion_kernel_runtime.h>
+
+namespace nvfuser::python {
+
+// direct_bindings/runtime.cpp contains Fusion and FusionExecutorCache.
+// TODO IrContainer and Kernel would be implemented here.
+
+namespace {
+
+void bindFusion(py::module& nvfuser) {
+  py::class_<FusionGuard>(nvfuser, "FusionGuard")
+      .def(
+          py::init<Fusion*>(),
+          py::arg("fusion"),
+          R"(
+Create a new FusionGuard to manage the active fusion context.
+
+A FusionGuard is a RAII-style guard that sets the active fusion context for the current scope.
+When the guard is created, it sets the provided fusion as the active fusion.
+When the guard is destroyed, it restores the previous fusion context.
+
+Parameters
+----------
+fusion : Fusion
+    The fusion to set as the active fusion context.
+
+Examples
+--------
+>>> fusion = Fusion()
+>>> with FusionGuard(fusion):
+...     # Define fusion operations here
+...     t0 = ops.add(x, y)
+...     # The fusion context is automatically restored when exiting the with block
+
+Notes
+-----
+- Only one fusion can be active at a time
+- The guard automatically handles saving and restoring the previous fusion context
+- It's recommended to use the guard in a with statement for automatic cleanup
+)");
+
+  // NOTE: manage, get_managed, get_managed_safe, stop_managing, has_managed are
+  // template functions. Pybind requires explicit template specialization.
```

Comment:
what is the comment talking about?

Reply:
```cpp
template <typename T>
void Fusion::manage(std::string key, T data)
```

These `Fusion` member functions are template functions. Templates do not exist in the python language. To bind these functions, you must instantiate a full (explicit) template specialization, which is basically a regular function.

Reply:
sorry I wasn't pointing at the right line. I see what you are trying to document about using instantiated templates. I was wondering why we are mentioning `manage, get_managed ...`, since none of those are used here.

Reply:
It is a comment to future me because managed `Fusion` data is used in scheduling.